### ISSUE TITLE
Random freeze/beachball when browsing SMB volumes

### ISSUE THREAD

**Issue by nitrostudios** [doublecmd/doublecmd#1561]
### Version

Double Commander
Version: 1.2.0 alpha
Revision: 11277
Commit: 17c4b7f46
Build date: 2024/03/31
Lazarus: 3.99.0.0
FPC: 3.2.2
Platform: aarch64-Darwin-cocoa
OS version: Mac OS X 12.6.6

### Description

I have a problem with random DC freeze/beachball when browsing the SMB volumes, sometimes more, sometimes less often. It's especially bad when I have opened tab with remote location and there is a copy operation in progress from some other client, then the whole interface freezes until the copy operation is completed. Does not happen in Finder or Forklift 4, but the same thing happens also in Nimble Commander, for example. Anyone else having the same problem?

**Comment by nitrostudios** [comment]
I've done some more testing and the it seems that only the application GUI freezes, because if there is an operation in progress (copy, move), it's still going on in the background, I can see it in the network monitor or disk activity. Can something like that depend on the API - Cocoa VS SwiftUI ? The Finder and Forklift 4 are using the SwiftUI.

**Comment by alexx2000** [comment]
I don't think so. Simply DC works with network folder same way as with local folder. Any network freeze will freeze DC.

**Comment by alex-s-team** [comment]
happens to me as well. can confirm that finder and forklift 4 don't have that problem. can also confirm that a network freeze triggers it - i.e. a directory with a (very) large number of files gets refreshed. maybe the multithreaded refresh / reload of a directory pane is handled differently?

**Comment by kaitrek** [comment]
Related to network freezes, is it possible to run each Pane in its own process so that when right pane is frozen accessing a remote location, the left pane can still browse?

**Comment by nitrostudios** [comment]
Just a small update: I've been in touch with Apple several times and developers are aware of the issue and are currently looking for more information. The problem seems to be with macOS SMB implementation. If there is anyone that is experiencing the same issue, please report it to the Apple support. I've been told that the more people will report the higher chance is for a fix to happen!

**Comment by chachizhu** [comment]
@alexx2000 
Hello
I found that making the following changes on macOS can reduce the likelihood of beachball occurrences:
	1.	In the ```Colors``` configuration, when matching file types, the default subfolder search depth is set to ```all```. This needs to be changed to ```current```.
	2.	In the ```Auto Refresh``` configuration, disable ```Auto Refresh``` for SMB mappings.

**Comment by alexx2000** [comment]
Hi,

1. This setting is not used by coloring function. So it should not has any effect.

**Comment by Valentin-N** [comment]
I can replicate the problem with my SMB shares, and I can confirm that adding exceptions for the SMB paths from auto-refresh mitigates the problem for me.

A similar issue that the OP created in https://github.com/mikekazakov/nimble-commander/issues/99 identified that calls to `fcntl` were likely problematic on MacOS. A quick search shows breadcrumbs pointing to issues with `fcntl` on SMB mount points for MacOS.

Given this, is the auto-refresh exception still the best mitigation for this issue?

**Comment by nitrostudios** [comment]
Well, I've abandoned the SMB on Mac and use NFS instead, a little bit more complicated, but at least it does not beachball every few minutes. I really doubt that Apple really cares about SMB at this point. 😢
